### PR TITLE
Fix: Error Fetching _devpagesmanifest.json #17274

### LINKED ISSUE #17274: Error fetching _devPagesManifest.json
# Bug report
I'm proxying my dev app from localhost to `https://` to avoid mixed request.
And when I hover on Next/Link nextjs tried to fetch _devPagesManifest.json but got rejected by SSL and then has error `pages is undefined`;

Could you add `credentials: 'same-origin'` to fetch options when loading `_devPagesManifest.json`?

### PR DESCRIPTION
**Pull request by itz-Me-Pj**
Closes NEXT-
Fixes #17274 Error Fetching _devpagesmanfest.json

### CODE REVIEW

**Review by timneutkens — DISMISSED** [review]
Hey, thanks for this PR, can you make sure a test is included that verifies what is described in the issue. Thanks!

**Comment by itz-Me-Pj** [comment]
> Hey, thanks for this PR, can you make sure a test is included that verifies what is described in the issue. Thanks!

Hi @timneutkens I can but can you provide a reference on how can I test this change or some other test I can refer to? As I am still exploring this project.

**Comment by timneutkens** [comment]
There's a guide for testing available here: https://github.com/vercel/next.js/blob/canary/contributing/core/testing.md

**Comment by itz-Me-Pj** [comment]
@timneutkens Added tests for the changes. Sorry it took a while. Feel free to mention if more changes are required

File: packages/next/src/client/page-loader.ts [hunk 1]
```diff
@@ -64,7 +64,7 @@ export default class PageLoader {
         return window.__DEV_PAGES_MANIFEST.pages
       } else {
         this.promisedDevPagesManifest ||= fetch(
-          `${this.assetPrefix}/_next/static/development/${DEV_CLIENT_PAGES_MANIFEST}`
+          `${this.assetPrefix}/_next/static/development/${DEV_CLIENT_PAGES_MANIFEST}`,{credentials:'same-origin'}
```
Comment (ijjk) [inline packages/next/src/client/page-loader.ts:67]: If we're applying this for the devPagesManifest, we should also apply it for the `devMiddlewareManifest` fetch and the stylesheets fetch as well 

x-ref: https://github.com/vercel/next.js/blob/e652a6f652c5f08ac87285e0f2140499103acd43/packages/next/src/client/page-loader.ts#L100-L102
x-ref: https://github.com/vercel/next.js/blob/e652a6f652c5f08ac87285e0f2140499103acd43/packages/next/src/client/route-loader.ts#L326

Comment (itz-Me-Pj) [reply packages/next/src/client/page-loader.ts:67]: ok @ijjk

**Comment by itz-Me-Pj** [comment]
@ijjk added all the changes requested

**Comment by ijjk** [comment]
## Failing test suites 
Commit: c2db198fb5420b575e62854e6b020be121c20e97

`pnpm test test/integration/dynamic-routing/test/index.test.js`
- Dynamic Routing > dev mode > should resolve dynamic route href for page added later
- Dynamic Routing > production mode > should output a routes-manifest correctly
- Dynamic Routing > production mode > should output a pages-manifest correctly

<details>
<summary>Expand output</summary>

  ● Dynamic Routing › dev mode › should resolve dynamic route href for page added later

    TIMED OUT: success

    undefined

    TypeError: Failed to parse URL from /_next/static/development/_devPagesManifest.json

      636 |
      637 |   if (hardError) {
    > 638 |     throw new Error('TIMED OUT: ' + regex + '\n\n' + content + '\n\n' + lastErr)
          |           ^
      639 |   }
      640 |   return false
      641 | }

      at check (lib/next-test-utils.ts:638:11)
      at Object.<anonymous> (integration/dynamic-routing/test/index.test.js:1159:7)

  ● Dynamic Routing › production mode › should output a routes-manifest correctly

    expect(received).toEqual(expected) // deep equality

    - Expected  - 0
    + Received  + 8

    @@ -53,10 +53,18 @@
            },
          },
        ],
        "dynamicRoutes": Array [
          Object {
    +       "namedRegex": "^/added\\-later/(?<nxtPslug>[^/]+?)(?:/)?$",
    +       "page": "/added-later/[slug]",
    +       "regex": "^\\/added\\-later\\/([^\\/]+?)(?:\\/)?$",
    +       "routeKeys": Object {
    +         "nxtPslug": "nxtPslug",
    +       },
    +     },
    +     Object {
            "namedRegex": "^/b/(?<nxtP123>[^/]+?)(?:/)?$",
            "page": "/b/[123]",
            "regex": "^\\/b\\/([^\\/]+?)(?:\\/)?$",
            "routeKeys": Object {
              "nxtP123": "nxtP123",

      1249 |       }
      1250 |
    > 1251 |       expect(manifest).toEqual({
           |                        ^
      1252 |         version: 3,
      1253 |         pages404: true,
      1254 |         caseSensitive: false,

      at Object.toEqual (integration/dynamic-routing/test/index.test.js:1251:24)

  ● Dynamic Routing › production mode › should output a pages-manifest correctly

    expect(received).toEqual(expected) // deep equality

    - Expected  - 0
    + Received  + 1

    @@ -7,10 +7,11 @@
        "/[name]/comments": "pages/[name]/comments.js",
        "/[name]/on-mount-redir": "pages/[name]/on-mount-redir.html",
        "/_app": "pages/_app.js",
        "/_document": "pages/_document.js",
        "/_error": "pages/_error.js",
    +   "/added-later/[slug]": "pages/added-later/[slug].html",
        "/another": "pages/another.html",
        "/b/[123]": "pages/b/[123].js",
        "/blog/[name]/comment/[id]": "pages/blog/[name]/comment/[id].js",
        "/c/[alongparamnameshouldbeallowedeventhoughweird]": "pages/c/[alongparamnameshouldbeallowedeventhoughweird].js",
        "/catchall-dash/[...hello-world]": "pages/catchall-dash/[...hello-world].html",

      1526 |       )
      1527 |
    > 1528 |       expect(manifest).toEqual({
           |                        ^
      1529 |         '/[name]/[comment]': 'pages/[name]/[comment].js',
      1530 |         '/[name]/[comment]/[...rest]': 'pages/[name]/[comment]/[...rest].js',
      1531 |         '/[name]/comments': 'pages/[name]/comments.js',

      at Object.toEqual (integration/dynamic-routing/test/index.test.js:1528:24)

</details>


Read more about building and testing Next.js in [contributing.md](https://github.com/vercel/next.js/blob/canary/contributing.md).

**Comment by ijjk** [comment]
# Stats from current PR

<details open>
<summary><strong>Default Build</strong></summary>

<br/>

<details>
<summary><strong>General</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | itz-Me-Pj/next.js sameOriginFix | Change |
| - | - | - | - |
| buildDuration | 10.7s | 10.8s | N/A |
| buildDurationCached | 5.9s | 5s | N/A |
| nodeModulesSize | 318 MB | 318 MB | ⚠️ +1.19 kB |
| nextStartRea..uration (ms) | 433ms | 440ms | N/A |

</details>

<details>
<summary><strong>Client Bundles (main, webpack)</strong></summary>

|  | vercel/next.js canary  | itz-Me-Pj/next.js sameOriginFix | Change |
| - | - | - | - |
| 3f784ff6-HASH.js gzip | 53.4 kB | 53.4 kB | ✓ |
| 423.HASH.js gzip | 185 B | 181 B | N/A |
| 68-HASH.js gzip | 29.7 kB | 29.7 kB | N/A |
| framework-HASH.js gzip | 45.2 kB | 45.2 kB | ✓ |
| main-app-HASH.js gzip | 238 B | 238 B | ✓ |
| main-HASH.js gzip | 31.8 kB | 31.8 kB | N/A |
| webpack-HASH.js gzip | 1.7 kB | 1.7 kB | ✓ |
| Overall change | 101 kB | 101 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | itz-Me-Pj/next.js sameOriginFix | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 31 kB | 31 kB | ✓ |
| Overall change | 31 kB | 31 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | itz-Me-Pj/next.js sameOriginFix | Change |
| - | - | - | - |
| _app-HASH.js gzip | 194 B | 195 B | N/A |
| _error-HASH.js gzip | 182 B | 181 B | N/A |
| amp-HASH.js gzip | 502 B | 501 B | N/A |
| css-HASH.js gzip | 320 B | 322 B | N/A |
| dynamic-HASH.js gzip | 2.5 kB | 2.5 kB | N/A |
| edge-ssr-HASH.js gzip | 255 B | 256 B | N/A |
| head-HASH.js gzip | 350 B | 349 B | N/A |
| hooks-HASH.js gzip | 368 B | 369 B | N/A |
| image-HASH.js gzip | 4.22 kB | 4.21 kB | N/A |
| index-HASH.js gzip | 257 B | 256 B | N/A |
| link-HASH.js gzip | 2.61 kB | 2.61 kB | N/A |
| routerDirect..HASH.js gzip | 310 B | 311 B | N/A |
| script-HASH.js gzip | 384 B | 383 B | N/A |
| withRouter-HASH.js gzip | 306 B | 308 B | N/A |
| 1afbb74e6ecf..834.css gzip | 106 B | 106 B | ✓ |
| Overall change | 106 B | 106 B | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | itz-Me-Pj/next.js sameOriginFix | Change |
| - | - | - | - |
| _buildManifest.js gzip | 484 B | 484 B | ✓ |
| Overall change | 484 B | 484 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | itz-Me-Pj/next.js sameOriginFix | Change |
| - | - | - | - |
| index.html gzip | 528 B | 527 B | N/A |
| link.html gzip | 539 B | 537 B | N/A |
| withRouter.html gzip | 523 B | 522 B | N/A |
| Overall change | 0 B | 0 B | ✓ |

</details>

<details>
<summary><strong>Edge SSR bundle Size</strong></summary>

|  | vercel/next.js canary  | itz-Me-Pj/next.js sameOriginFix | Change |
| - | - | - | - |
| edge-ssr.js gzip | 94 kB | 94 kB | N/A |
| page.js gzip | 149 kB | 149 kB | N/A |
| Overall change | 0 B | 0 B | ✓ |

</details>

<details>
<summary><strong>Middleware size</strong></summary>

|  | vercel/next.js canary  | itz-Me-Pj/next.js sameOriginFix | Change |
| - | - | - | - |
| middleware-b..fest.js gzip | 624 B | 623 B | N/A |
| middleware-r..fest.js gzip | 151 B | 149 B | N/A |
| middleware.js gzip | 47.4 kB | 47.4 kB | N/A |
| edge-runtime..pack.js gzip | 1.94 kB | 1.94 kB | ✓ |
| Overall change | 1.94 kB | 1.94 kB | ✓ |

</details>

<details>
<summary><strong>Next Runtimes</strong></summary>

|  | vercel/next.js canary  | itz-Me-Pj/next.js sameOriginFix | Change |
| - | - | - | - |
| app-page-exp...dev.js gzip | 166 kB | 166 kB | ✓ |
| app-page-exp..prod.js gzip | 95.1 kB | 95.1 kB | ✓ |
| app-page-tur..prod.js gzip | 96.9 kB | 96.9 kB | ✓ |
| app-page-tur..prod.js gzip | 91.5 kB | 91.5 kB | ✓ |
| app-page.run...dev.js gzip | 135 kB | 135 kB | ✓ |
| app-page.run..prod.js gzip | 90 kB | 90 kB | ✓ |
| app-route-ex...dev.js gzip | 22 kB | 22 kB | ✓ |
| app-route-ex..prod.js gzip | 14.8 kB | 14.8 kB | ✓ |
| app-route-tu..prod.js gzip | 14.8 kB | 14.8 kB | ✓ |
| app-route-tu..prod.js gzip | 14.6 kB | 14.6 kB | ✓ |
| app-route.ru...dev.js gzip | 21.7 kB | 21.7 kB | ✓ |
| app-route.ru..prod.js gzip | 14.6 kB | 14.6 kB | ✓ |
| pages-api-tu..prod.js gzip | 9.43 kB | 9.43 kB | ✓ |
| pages-api.ru...dev.js gzip | 9.7 kB | 9.7 kB | ✓ |
| pages-api.ru..prod.js gzip | 9.43 kB | 9.43 kB | ✓ |
| pages-turbo...prod.js gzip | 22 kB | 22 kB | ✓ |
| pages.runtim...dev.js gzip | 22.7 kB | 22.7 kB | ✓ |
| pages.runtim..prod.js gzip | 22 kB | 22 kB | ✓ |
| server.runti..prod.js gzip | 49.7 kB | 49.7 kB | ✓ |
| Overall change | 922 kB | 922 kB | ✓ |

</details>

<details>
<summary><strong>Diff details</strong></summary>

<details>
<summary>Diff for <strong>main-HASH.js</strong></summary>

Diff too large to display
</details>

</details>

</details>
Commit: c2db198fb5420b575e62854e6b020be121c20e97

**Review by ijjk — APPROVED** [review]
Thanks!

**Pull request merged**